src/wallet/rpcwallet.cpp

```diff
@@ -1731,23 +1755,55 @@ UniValue listsinceblock(const JSONRPCRequest& request)
         filter = filter | ISMINE_WATCH_ONLY;
     }
 
+    bool include_removed = (request.params.size() < 4 || request.params[3].get_bool());
+
     int depth = pindex ? (1 + chainActive.Height() - pindex->nHeight) : -1;
 
     UniValue transactions(UniValue::VARR);
+    std::set<uint256> listed;
 
     for (map<uint256, CWalletTx>::iterator it = pwalletMain->mapWallet.begin(); it != pwalletMain->mapWallet.end(); it++)
     {
         CWalletTx tx = (*it).second;
 
         if (depth == -1 || tx.GetDepthInMainChain() < depth)
-            ListTransactions(tx, "*", 0, true, transactions, filter);
+        {
+            if (ListTransactions(tx, "*", 0, true, transactions, filter) && include_removed)
+            {
+                listed.insert(it->first);
+            }
+        }
+    }
+
+    // when a reorg'd block is requested, we also list any relevant transactions
+    // in the blocks of the chain that was detached
+    UniValue removed(UniValue::VARR);
+    while (include_removed && paltindex && paltindex != pindex)
+    {
+        CBlock block;
+        if (!ReadBlockFromDisk(block, paltindex, Params().GetConsensus()))
+        {
+            throw JSONRPCError(RPC_INTERNAL_ERROR, "Can't read block from disk");
```

Comment:
Yea, I was heasitant to recommend doing it alone because I'm not confident about how safe it is to make sure the wallet's data is always right, will have to think more and review once you have it coded :).